tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -0,0 +1,116 @@
+﻿// <copyright file="BlockingMiddleware.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NETFRAMEWORK
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Transports;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class BlockingMiddleware
+{
+    private readonly bool _runSecurity;
+    private readonly bool _endPipeline;
+    private readonly RequestDelegate? _next;
+
+    internal BlockingMiddleware(RequestDelegate? next = null, bool runSecurity = false, bool endPipeline = false)
```

Comment:
Do we ever set `runSecurity = false`? 🤔  AFAICT, we only added it in the `ApplicationBuilder` integration, and we set `runSecurity=true` always there?

Reply:
good point, indeed we don't! at first I added the blocking middleware every other middleware, and wanted it to run security only at the beginning and the end, as the other stages are covered by integrations where we get mvc informations we need. But I'll remove it for now, and see later when we deal with the `BlockException`